R/board_rsconnect.R

```diff
@@ -549,20 +550,77 @@ rsc_user_name <- function(board, guid) {
 }
 
 read_cache <- function(path) {
-  if (file.exists(path)) {
+  if (file.exists(path) && use_connect_cache()) {
     yaml::read_yaml(path, eval.expr = FALSE)
   } else {
     list()
   }
 }
-update_cache <- function(path, key, value) {
-  cache <- read_cache(path)
-  cache[[key]] <- value
-  write_yaml(cache, path)
 
+update_cache <- function(path, key, value) {
+  if (use_connect_cache()) {
+    cache <- read_cache(path)
+    cache[[key]] <- value
+    write_yaml(cache, path)
+  }
   value
 }
 
+#' @export
+#' @rdname use_connect_cache
+clear_connect_cache <- function(board,
+                                cache = c("content-cache.yml", "users-cache.yml")) {
+  path <- fs::path(board$cache, cache)
+  cli::cli_inform("Deleting {.file {path}}")
+  fs::file_delete(path)
+}
+
+#' Detect whether to use content and user caches on RStudio Connect
+#'
+#' Sometimes the content and user caches for RStudio Connect can get out of
+#' sync with the server. While we work on solving this problem, the function
+#' `use_connect_cache()` helps you by detecting whether or not to use these
+#' caches. You can manually delete a broken cache via `clear_connect_cache()`.
+#'
+#' @details The `use_connect_cache()` function uses the following method for
+#' detecting whether or not to use the content and user caches for Connect:
+#'
+#' 1. If the `pins_connect_cache` option is set to `FALSE`, `FALSE` is returned.
+#' 2. If the `PINS_CONNECT_CACHE` environment variable is set to `"false"`,
+#' `"FALSE"`, or `"False"`, `FALSE` is returned
+#' 3. In all other cases (including empty or `NULL` values for either of the
+#' above), `TRUE` is returned
```

Comment:
The logic in the function is more complicated in order to give some helpful error messages for bad values. Although maybe I should reword this, in that case?